connectors/translator-swagger/src/main/java/org/teiid/translator/swagger/SwaggerMetadataProcessor.java

```diff
@@ -0,0 +1,286 @@
+    private static final String PARAMETER_SEPARATOR_EQUAL = "="; //$NON-NLS-1$
+    private static final String IS_PATH_PARAMETER = "isPathParam"; //$NON-NLS-1$
+    private static final String BASE_URL = "restBaseUrl"; //$NON-NLS-1$
+    private static final String HTTP_HOST = "httpHost"; //$NON-NLS-1$
+    private static final String HTTP_ACTION = "httpAction"; //$NON-NLS-1$
+    private static final String HTTP_API_PATH = "restApiPath"; //$NON-NLS-1$
+    
+    private static final String TRUE_STRING = "true"; //$NON-NLS-1$
+    private static final String FALSE_STRING = "false"; //$NON-NLS-1$
+    
+    private SwaggerExecutionFactory ef;
+    private String httpHost = "http://localhost:8080"; //$NON-NLS-1$
+    private String baseUrl = "/"; //$NON-NLS-1$
+    
+    protected void setExecutionfactory(SwaggerExecutionFactory ef) {
+        this.ef = ef;
+    }
+    
+    @Override
+    public void process(MetadataFactory mf, WSConnection connection) throws TranslatorException {
+        
+        Swagger swagger = getSchema(connection);
+        baseUrl = swagger.getBasePath();
+        if(swagger.getHost() != null && !swagger.getHost().equals("")) { //$NON-NLS-1$
+            //TODO-- use other scheme
+            String scheme = "http"; //$NON-NLS-1$
+            if(swagger.getSchemes().size() > 0) {
+                scheme = swagger.getSchemes().get(0).toValue();
+            }
+            httpHost = scheme + "://" + swagger.getHost();//$NON-NLS-1$
+        }
+        
+        for(Entry<String, Path> entry : swagger.getPaths().entrySet()) {
+            addProcedure(mf, entry.getKey(), entry.getValue());
+        }
+    }
+    
+    private void addProcedure(MetadataFactory mf, String key, Path path) {
+        
+        for(Entry<HttpMethod, Operation> entry : path.getOperationMap().entrySet()){
+            
+            String action = entry.getKey().toString();
+            
+            Operation operation = entry.getValue();
+            String procName = operation.getOperationId();
+            String produces = getProduceTypes(operation);
+            String consumes = getConsumeTypes(operation);
+            String annotation = getOperationSummary(operation);
+            
+            Procedure procedure = mf.addProcedure(procName);
+            procedure.setVirtual(false);
+            procedure.setNameInSource(procName);
+            procedure.setProperty(HTTP_ACTION, action);
+            procedure.setProperty(HTTP_API_PATH, key);
+            procedure.setProperty(BASE_URL, baseUrl);
+            procedure.setProperty(HTTP_HOST, httpHost); 
+            procedure.setProperty(PROCEDURE_PRODUCES, produces);
+            procedure.setProperty(PROCEDURE_CONSUMES, consumes);
+            procedure.setAnnotation(annotation);
+            
+            mf.addProcedureParameter("result", TypeFacility.RUNTIME_NAMES.BLOB, Type.ReturnValue, procedure); //$NON-NLS-1$ 
```

Comment:
We can NOT have blob as return, that is what we did before when we did not knew what is being returned. The value of Swagger is it defines what is being returned. You have "ref" element pointing to a data structure, that needs to be return of this procedure. If the blob is the return the user still needs to write parsing logic using XMLTABLE in a view transformation. There we are not providing much value, other than just providing SQL way execute the REST API.

This is what I was saying before, you have two choices here. One to create virtual procedure, that uses the XMLTABLE semantics to parse the payload into a tabular structure, or write parser for JSON and XML. Since Swagger tell how these payloads are formatted it should be easy enough to do either. If the payload type is unknown or not a JSON/XML then use the blob.